docs/PasswordManagement.md

```diff
@@ -35,32 +38,119 @@ Note: Before the creation of any Splunk deployments in the kubernetes namespace,
+
+Below Example explains the steps required for setting up IAM Service Account
+
+- Follow the steps defined [here to create IAM Role for Service Account](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html)
+
+- Make sure IAM Role only has least amount of privilege necessary for smartstore to work.
+
+- Make sure the service account is used in custom resources where its required
+ Once the Service Account is created, make sure it is annotated with specific IAM Role. Once everything looks good, add service account to splunk custom resource. here is the example for adding it to `Standalone` instance
+
+
+```
+apiVersion: enterprise.splunk.com/v4
+kind: Standalone
+metadata:
+  name: test
+spec:
+  serviceAccount: oidc-service-account
+  smartstore:
+    defaults:
+        volumeName: test-cluster-bucket
+    indexes:
+      - name: main
+        remotePath: $_index_name
+        volumeName: test-cluster-bucket
+      - name: cloudwatch
+        remotePath: $_index_name
+        volumeName: test-cluster-bucket
+    volumes:
+      - name: test-cluster-bucket
+        path: test-cluster-bucket/smartstore
+        endpoint: https://s3-us-west-2.amazonaws.com
+```
+
+- Make sure the IAM service account is used only in required custom resources
+
+- When Splunk pod is running AWS webhook service injects 2 new environment variables `AWS_WEB_IDENTITY_TOKEN_FILE` and `AWS_ROLE_ARN` along with JWS Token file. `splunk` pod reads these environment variables to get temporary AWS credentials from AWS IAM service to access smartstore buckets
+
+***OIDC key management***
+The proper Key management of OIDC is outside of Splunk installation. The customer is responsible to use a properly configured OIDC using certificates from a trusted CA.
+
+***Self signed certificate***
+The OIDC should not use self-signed certificates but rather utilize an existing PKI infrastructure, e.g. have the OIDC certificate issued and signed by your organization's CA with proper certificate signature chains and key expiation policies.
+
+***Sharing OIDC token file***
+Make sure the token file mentioned in AWS_WEB_IDENTITY_TOKEN_FILE location is only accessible inside of the pod and is not mapped or shared outside of the pod
+
+## Support for AWS IAM Role for Service Account in Splunk Operator Deployment
+
+Follow the steps mentioned above for creating AWS IAM Service Account. Make sure IAM Role only has least amount of privilege necessary reading apps from S3 bucket. Once the service account is created, map this service account to `splunk-operator` deployment. Below is the example
+
+```
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: splunk-operator-controller-manager
+  namespace: splunk-operator
+spec:
+  progressDeadlineSeconds: 600
+  replicas: 1
+  revisionHistoryLimit: 10
+  ...
+  spec:
+    containers:
+    -
+      ...
+      serviceAccount: oidc-service-account
```

Comment:
Nit: Line 148 has `serviceAccount` and 149 has `serviceAccountName`. Do we need both?

Reply:
yes we need both that is an example, by default it adds serviceAccountName